src/software/ai/evaluation/find_passes.cpp

```diff
@@ -0,0 +1,101 @@
+#include "software/ai/evaluation/find_passes.h"
+
+Circle getObstacle(const Robot& robot, double radius)
+{
+    return Circle(robot.position(), radius);
+}
+
+std::vector<Circle> getObstacles(const std::vector<Robot>& robots, double radius)
+{
+    std::vector<Circle> obstacles;
+    for (Robot robot : robots)
+    {
+        obstacles.push_back(getObstacle(robot, radius));
+    }
+    return obstacles;
+}
+
+std::vector<Robot> findOpenFriendlyRobots(const Team& friendly_team,
+                                          const Team& enemy_team, double radius)
+{
+    std::vector<Circle> obstacles = getObstacles(enemy_team.getAllRobots(), radius);
+
+    std::vector<Robot> open_robots;
+    for (Robot friendly : friendly_team.getAllRobots())
+    {
+        if (std::all_of(obstacles.begin(), obstacles.end(),
+                        [friendly](Circle enemy_circle) {
+                            return !contains(enemy_circle, friendly.position());
+                        }))
+        {
+            open_robots.push_back(friendly);
+        }
+    }
+    return open_robots;
+}
+
+std::vector<Pass> findDirectPasses(const Robot& robot, const Team& friendly_team,
+                                   const Team& enemy_team)
+{
+    std::vector<Robot> open_robots =
+        findOpenFriendlyRobots(friendly_team, enemy_team, ROBOT_MAX_RADIUS_METERS * 1.5);
+
+    open_robots.erase(remove(open_robots.begin(), open_robots.end(), robot));
+    std::vector<Robot> robot_obstacles = friendly_team.getAllRobots();
+    robot_obstacles.insert(robot_obstacles.begin(), enemy_team.getAllRobots().begin(),
+                           enemy_team.getAllRobots().end());
+    robot_obstacles.erase(remove(robot_obstacles.begin(), robot_obstacles.end(), robot));
+
+    std::vector<Circle> obstacles =
+        getObstacles(robot_obstacles, ROBOT_MAX_RADIUS_METERS * 1.5);
+    std::vector<Pass> direct_passes;
+    for (Robot open_robot : open_robots)
+    {
+        Circle obstacle = getObstacle(open_robot, ROBOT_MAX_RADIUS_METERS * 1.5);
```

Comment:
Where should this constant be stored?

Reply:
if you take the suggestion for a findAllPasses function then it can be a default argument, e.g. `findAllPasses(..., radius= ROBOT_MAX_RADIUS_METERS * 1.5 )`

Reply:
Fixed